src/lib/ppx_version/versioned_util.ml

```diff
@@ -74,3 +74,37 @@ let jane_street_library_modules = [ "Uuid" ]
 
 let jane_street_modules =
   [ "Core"; "Core_kernel" ] @ jane_street_library_modules
+
+module StringSet = Set.Make (String)
+
+let collect_types : StringSet.t Ast_traverse.fold =
+  let rec lident l acc =
+    match l with
+    | Lident s ->
+        StringSet.add acc s
+    | Ldot (m, s) ->
+        lident m (StringSet.add acc s)
```

Comment:
Shouldn't this be `lident m acc`? We don't care about the RHS of the path at all IMO

Reply:
Of course, this allows aliasing á la
```ocaml
module Make (A : S) = struct
  module B = A
  module C = struct
    module%versioned Stable = struct
      module V1 = struct
        type t = B.t * B.t
      end
    end
  end
end
```
so this really circumvents the protection. We should move all the versioning out of the main code anyway, so I don't really care at this point.

Reply:
> Of course, this allows aliasing á la ...

yeah I also thought of this :disappointed: 

> We should move all the versioning out of the main code anyway, so I don't really care at this point.

I'm assuming here you mean if you want to defined versioned types, they need to be in a standalone module that contains only type definitions? This would be nice since this system doesn't always compose well with vanilla ocaml, maybe a lot of work.

In the meanwhile, what is your most preferred/least hated option:

1. Something like this PR and remain vigilant
2. Remove the `in_functor` check entirely and remain more vigilant
3. Separate out type definitions piecemeal, e.g. if I'm already working in there anyway